web/packages/teleterm/src/ui/services/workspacesService/accessRequestsService.ts

```diff
@@ -258,6 +299,7 @@ export type ResourceRequest =
       kind: 'kube';
       resource: {
         uri: KubeUri;
+        namespaces?: KubeResourceNamespaceUri[];
```

Comment:
This is unused, no?

But I wonder if it wouldn't be more convenient to keep the selected namespaces here, instead of having them as a separate resource (line 313).
Pros:
1. We wouldn't need to remember about removing `kube_cluster` resources (from UI/API requests) when there are namespaces selected. 
2. We wouldn't have to remove namespaces manually when a "parent" kube_cluster is removed from the access request.

Cons:
1. This would differ from shared types where the namespace is a regular requestable resource. OTOH I feel that the proposed data model is more correct.

I can imagine having a function like this one in `AccessRequestsService`:
```ts
  async addOrRemoveKubeNamespace(resourceUri: KubeResourceNamespaceUri) {
    this.setState(draftState => {
      if (draftState.pending.kind !== 'resource') {
        throw new Error('Cannot add a kube namespace to a role access request');
      }

      const { resources } = draftState.pending;

      const requestedResource = resources.get(
        routing.getKubeUri(
          routing.parseKubeResourceNamespaceUri(resourceUri).params
        )
      );
      if (!requestedResource || requestedResource.kind !== 'kube') {
        throw new Error('Cannot add a kube namespace to a non-kube resource');
      }
      const kubeResource = requestedResource.resource;
      if (!kubeResource.namespaces) {
        kubeResource.namespaces = new Set();
      }
      if (kubeResource.namespaces.has(resourceUri)) {
        kubeResource.namespaces.delete(resourceUri);
      } else {
        kubeResource.namespaces.add(resourceUri);
      }
    });
  }
```
Please let me know what do you think about it :)

Reply:
i think it's great! definitely cleaned up some code: https://github.com/gravitational/teleport/pull/47347/commits/2042699afcb31ee76831b5785f6287d2101b007b